packages/eth-edge/src/controllers/wallets.controller.ts

```diff
@@ -14,4 +17,10 @@ export class WalletsController {
         const createWalletReq = await useValidate(createWalletSchema, body);
         return this.walletsService.create(createWalletReq);
     }
+
+    @Get("users/:id/wallets")
+    async findAll(@Param() params: GetWalletsRequest): Promise<Wallet[]> {
+        const getWalletReq = await useValidate(getWalletsSchema, params);
+        return await this.walletsService.findAll(getWalletReq.id);
```

Comment:
Don't need to await when you return a value, it messes with nodes event loop and the ordering of things, best practice is to let the caller await the value since the function is wrapping it in a promise again anyways

```suggestion
        return this.walletsService.findAll(getWalletReq.id);
```